src/wallet/rpcwallet.cpp

```diff
@@ -4971,6 +4979,17 @@ static RPCHelpMan walletcreatefundedpsbt()
         }
     }
     assert(blinder_index < rawTx.vin.size()); // We added inputs, or existing inputs are ours, we should have a blinder index at this point.
+    // It may add outputs (change, and in some edge case OP_RETURN) which need to be
+    // blinded. So pull these into `psbt_outs`.
+    for (auto it = rawTx.vout.begin(); it != rawTx.vout.end(); ++it) {
+        if (!it->nNonce.IsNull() && !psbt_outs.count(*it)) {
+            PSBTOutput new_out{2}; // psbtv2 output
+            new_out.m_blinding_pubkey.Set(it->nNonce.vchCommitment.begin(), it->nNonce.vchCommitment.end());
+            new_out.m_blinder_index = blinder_index;
+            psbt_outs.insert(std::make_pair(*it, new_out));
+        }
+    }
```

Comment:
In 70faa7e6b7a4967302154c51a1898d173e594f15 "walletcreatefundedpsbt: signal blinding data correctly to `FundTransaction`"

nit: This could be a range based for loop:

```suggestion
    for (const CTxOut& txout : rawTx.vout) {
        if (!out.nNonce.IsNull() && !psbt_outs.count(txout)) {
            PSBTOutput new_out{2}; // psbtv2 output
            new_out.m_blinding_pubkey.Set(txout.nNonce.vchCommitment.begin(), txout.nNonce.vchCommitment.end());
            new_out.m_blinder_index = blinder_index;
            psbt_outs.insert(std::make_pair(txout, new_out));
        }
    }
```